src/types.ts

```diff
@@ -137,6 +162,190 @@ type FinalAuthReqsSatisfied<T extends HasFinalAuthorize<any, any>> = IntersectPr
 */
 // Start with the "optionalized" section below
 
```

Comment:
AFAICT ObjectValueToAny doesn't make any sense if you don't pass it an object for T.  In fact, Record<> won't make sense on a non-object anyway.
So
- [ ] instead of the conditional, type-constrain it.